index.html

```diff
@@ -301,111 +301,78 @@
-        </li>
-        <li>
-          Protect the integrity of <a>Thing</a>s and <a>Thing Description</a>s.
-        </li>
-      </ul>
-    </p>
-    </section>
-
-    <section> <h3>Security Mechanisms</h3>
-      <section> <h4>Identification, Authentication, Authorization</h4>
-      </section>
-
-      <section> <h4>Transport-level security</h4>
-      </section>
-
-      <section> <h4>Application-level security</h4>
-      </section>
-    </section>
-
-    <section class="informative"> <h3>Security policies</h3>
-      <p>
-        This section summarizes the security policies which are specified as
-        normative requirements in the respective algorithms of this
-        specification.
-      </p>
-    </section>
-  </section>
-
   <section> <h2>WoT Data Representation</h2>
   <p>
   </p>
   </section>
 
   <section> <h2>The WoT API</h2>
   <p>
-    The API object provides access obtain <a>Thing</a>s by discovery or creation. A <a>Thing</a> is represented either as a <a>ConsumedThing</a> or <a>ExposedThing</a>.
+    The API object represents an implementation of the <a>WoT Runtime</a> and provides functionality to obtain <a>Thing</a>s by discovery or creation.
   </p>
   <p>
     The <dfn>WoT object</dfn> exposes only functions and has no internal state.
   </p>
-  <p class="issue">
-     Because of this, implementations MAY use a global object, or provide a constructor, or provide the API object, or by other means (e.g. by `require()`). Since the common denominator across various <a>Runtime</a> types would be to use a constructor, this would be recommended. As a consequence, the <a>ConsumedThing</a> and <a>ExposedThing</a> objects could also be constructed, complete with an asynchronous fetch functionality.
-  </p>
   <p class="note">
-    The WebIDL interfaces presented in this initial version of the document are based on the [[!WOT-PRACTICES]]. It is expected to be updated soon.
+     Browser implementations SHOULD use a namespace object such as `wot`, and [Node.js](https://nodejs.org/en/)-like runtimes MAY provide the API object through the [`require()`](https://nodejs.org/api/modules.html) mechanism.
   </p>
   <pre class="idl">
+    // [SecureContext]
+    // [NamespaceObject]
     interface WoT {
-      Promise&lt;void&gt; discover(ThingFilter filter, ThingDiscoveryCallback onfound);
-
-      Promise&lt;ConsumedThing&gt; retrieve((USVString or Dictionary) thingReference);
-
-      Promise&lt;ExposedThing&gt; createThing(ThingInit init);
+      Observable&lt;ConsumedThing&gt; discover(optional ThingFilter filter);
+      Promise&lt;ConsumedThing&gt; retrieve(USVString url);
+      Promise&lt;ExposedThing&gt; createLocalThing(ThingInit init);
     };
 
     dictionary ThingInit {
       DOMString name;
       USVString url;
-      object description;
+      Dictionary description;
     };
 
-    callback ThingDiscoveryCallback = void (ConsumedThing thing);
+    enum DiscoveryType { "any", "local", "directory", "broadcast" };
```

Comment:
No, enums are not extensible, if we wanted that would be just using a String.
However, this was intentional, as the supported discovery types should be standardized, since they require explicit and well defined support from the underlying platform and depend on well known standards.
There is no problem adding them to the spec, but they should be explicitly specified. Subsequent versions of the spec can extend the enum with new values.
Otherwise the API would be like `discover({ type: "app-specific-opaque-string"});` which is more like a free text search than scoping for supported discovery types.